sys/crypto/chacha.c

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (C) 2015 Freie Universität Berlin, Computer Systems & Telematics
```

Comment:
> no and I am not interrested in the copyright at all. do I need to add one?

Concerning German law this line is pretty useless anyway - your the author and therefore copyright holder. For US other foreign laws, I don't know what happens if you omit this line. If it doesn't bother you, I think it's easiest if you just put your name here. (Of course, you can also grant it also to someone else - or even to the university. At least for US law.)

> (when I did some code based on older or already written code - i ll just leave it yes?)

Yes, but you are allowed to add your own copyright.